spec/tasks/support/gem_base64_spec.rb

```diff
@@ -0,0 +1,77 @@
+require "tasks/support/gem_base64"
+require "manageiq_performance/version"
+require "fileutils"
+
+def build_gem file
+  pkg_dir = File.dirname file
+  FileUtils.mkdir_p pkg_dir
+  FileUtils.touch file
+  file_obj = File.new(file, "w+")
+
+  Gem::Package.new(file_obj)
+              .tap{ |p| p.spec = GemBase64.gemspec }
+              .build
+end
+
+def decode_and_untar gem_string, pattern="*"
+  # Convert from a base64 string to a tar ball file contents
+  gem_io = StringIO.new Base64.decode64(gem_string)
+  # Un tar the specific file for comparison
+  extracted_contents = ""
+  gem_tar = Gem::Package::TarReader.new(gem_io)
+  gem_tar.each do |entry|
+    next unless entry.full_name == 'data.tar.gz'
+
+    Zlib::GzipReader.wrap entry do |gzio|
+      Gem::Package::TarReader.new(gzio).each do |dot_gem_file|
+        next unless File.fnmatch pattern, dot_gem_file.full_name, File::FNM_DOTMATCH
+        extracted_contents = dot_gem_file.read
+        # don't break here to avoid zlib warning
+      end
+    end
+
+    break # ignore further entries (optimization)
+  end
+
+  extracted_contents
+end
+
+describe GemBase64 do
+  let(:version) { Gem::Version.new(ManageIQPerformance::VERSION) }
+  let(:spec_dir) { File.expand_path "../..", File.dirname(__FILE__) }
+  let(:spec_tmp) { "#{spec_dir}/tmp" }
+  let(:pkg_dir)  { "#{spec_tmp}/pkg" }
+  let(:gem_file) { "#{pkg_dir}/manageiq_performance-#{version.to_s}.gem" }
+
+  around(:each) do |example|
+    Gem::DefaultUserInteraction.use_ui(Gem::SilentUI.new) do
+      build_gem gem_file
+      example.run
+    end
+    FileUtils.rm_rf spec_tmp
+  end
+
+  describe "::gemspec" do
+    it "reads the .gemspec file and returns a Gem::Specification instance" do
+      expect(GemBase64.gemspec.name).to    eq "manageiq-performance"
+      expect(GemBase64.gemspec.version).to eq version
+    end
+  end
+
+  describe "::gem_as_tar_io" do
+    it "returns an io object of the .gem (tar) contents" do
+      result = GemBase64.gem_as_tar_io
+      expect(result.read).to eq File.read(gem_file)
+    end
+  end
+
+  describe "::gem_as_base64_string" do
+    it "returns a base65 string version of the .gem that recompiles back correctly" do
+      bin_file = File.expand_path("../../../../bin/miqperf", __FILE__)
```

Comment:
should these be living in `exe` instead of `bin`?
Thought other gook ended up finding its way into `bin` via bundler
(requires changing bindir in gemspec)